src/pip/_internal/download.py

```diff
@@ -598,6 +598,9 @@ def __init__(self, *args, **kwargs):
     def add_insecure_host(self, host):
         # type: (str) -> None
         self.mount('https://{}/'.format(host), self._insecure_adapter)
+        if not urllib_parse.urlparse('https://{}/'.format(host)).port:
```

Comment:
Sorry for the delay. Can you make this line into a unit-testable function? It can be something like `does_netloc_have_port(netloc)` and be put before `split_auth_from_netloc()`. It will be easy to test that way, and you can follow the example of how the related functions are tested. It would be easy to include IPv6 tests so people can see what happens to those (as well as other edge cases).

Reply:
Change as requested.